src/embind/emval.js

```diff
@@ -222,6 +222,8 @@ var LibraryEmVal = {
       $$$embind_global$$$ = global;
     } else if (typeof window === 'object' && testGlobal(window)) {
       $$$embind_global$$$ = window;
+    } else if (typeof self === 'object' && testGlobal(self)) {
+      $$$embind_global$$$ = self;
```

Comment:
I wouldn't have guessed that `self` was for workers, though that might just be me. Might be worth a comment here?

Reply:
I agree. I've just added it for the sake of clarity.

Reply:
It would be good to know how far back `self` works. caniuse isn't clear: https://caniuse.com/#feat=mdn-api_window_self

The safe thing might be to check `ENVIRONMENT_IS_WEB` vs `_WORKER`, at least in `LEGACY_VM_SUPPORT` mode.

Reply:
@kripken I've added `ENVIRONMENT_IS_WORKER` precondition as a safety measure.

Reply:
`self` has existed basically forever (like `top`, `parent`, etc.). In fact, we don't even need a separate branch for `window` as `self` is more universal and works both in windows and in workers.

Reply:
Ok, sgtm to do it that way, thanks @RReverser 

@osman-turan sorry for the back-and-forth! But sounds like it'll be better to do it like that.